src/background/services/openPayments.ts

```diff
@@ -459,6 +464,21 @@ export class OpenPaymentsService {
     await this.storage.setState({ out_of_funds: false });
   }
 
+  async updateBudget({ amount, recurring }: UpdateBudgetPayload) {
+    const { walletAddress, ..._existingGrants } = await this.storage.get([
+      'walletAddress',
+      'oneTimeGrant',
+      'recurringGrant',
+    ]);
+
+    await this.completeGrant(
+      amount,
+      walletAddress!,
+      recurring,
+      InteractionIntent.BUDGET_UPDATE,
+    );
```

Comment:
TODO:
- complete new grant
- on success, revoke existing grants (both recurring and one-time)
- on failure, keep existing grants
- reflect progress/errors into popup